app/src/main/java/com/ubergeek42/WeechatAndroid/relay/BufferList.kt

```diff
@@ -385,13 +385,12 @@ object BufferList {
     private class LineListingHandler(private val bufferPointer: Long) : HdataHandler {
         override fun handleMessage(obj: Hdata, id: String) {
             findByPointer(bufferPointer)?.let { buffer ->
-                val newLines = ArrayList<Line>(obj.count)
 
+                buffer.clearLines()
```

Comment:
this can creates a (very rare) situation where the UI sees the buffer as empty. `replaceLines` makes the update atomic.

Reply:
I could add  the lines before replacing the list, would that be better for you?

Reply:
if you mean do it the way `replaceLies` did it, yeah, that would be better :p

Reply:
ok, will work on it.

Reply:
> `replaceLines` makes the update atomic.

This is actually only true for `unfiltered`, whereas for `filtered` they are added in a loop: https://github.com/ubergeek42/weechat-android/blob/master/app/src/main/java/com/ubergeek42/WeechatAndroid/relay/Lines.kt#L95

Assuming that we want to be consistent here, can you say which way you want it?

Reply:
it's effectively atomic because Buffer.replaceLines is synchronized; operations such as addAll are probably not atomic either way